### ISSUE TITLE
Broken in fish, version 3.1.1

### ISSUE THREAD

**Issue by amir20** [junegunn/fzf#2008]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [ ] Linux
    - [x] Mac OS X 
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [ ] zsh
    - [x] fish
    
## Problem / Steps to reproduce

I recently updated to `fish, version 3.1.1`. With this version, fzf seems to be broken. While investigating, I found that fzf correctly blocks to show data but no actual data is shown on the screen. 

I can reproduce this `100%` by doing `eval (__fzfcmd)`. 
<img width="1442" alt="Screen Shot 2020-04-27 at 1 21 57 PM" src="https://user-images.githubusercontent.com/260667/80417139-1fe22480-888a-11ea-9de3-ecf70ae12cdc.png">

However, if I do `up-arrow` then `enter` it does pick a value from the menu. It just seems hidden. 

Does anybody else have this issue? 

I am also using iTerm.

**Comment by nzw0301** [comment]
I have the same issue. `Ctr-r` does not show anything.

**Comment by vrinek** [comment]
Same issue here, same versions, also on macOS.

I noticed that something like `ls | fzf` works fine.

Hope this helps.

**Comment by amir20** [comment]
I changed my search function in fish to 

```
function __fzf_reverse_isearch
    history merge
    history -z | fzf --read0 | read -lz result
    and commandline -- $result
    commandline -f repaint
end
```

For now, at least I have a working search. But does seem like something has broken `eval` in latest Fish.

**Comment by breml** [comment]
From https://fishshell.com/release_notes.html about fish 3.1.1 release:

> Vi mode now supports R to enter replace mode (#6342)

So it could be related to the key binding of <kbd>Ctrl+R</kbd> when the closed https://github.com/fish-shell/fish-shell/pull/6342.

**Comment by breml** [comment]
Or it might be related to

> fzf in complex pipes no longer hangs. More generally, code run as part of command substitutions or eval will no longer have separate process groups. (#6624, #6806).

(also from the release notes)

**Comment by OakNinja** [comment]
This issue is adressed and will be fixed in 3.1.2 according to https://github.com/fish-shell/fish-shell/issues/6955#

**Comment by OakNinja** [comment]
Verified working in 3.1.2 - this issue may be closed.
